File: Data Structures & Algorithms/pow-x-n/submission-1.py
class Solution:
    def myPow(self, x: float, n: int) -> float:

        def recursion(x, n):
            if x == 0:
                return 0
            if n == 0:
                return 1
            res = recursion(x * x, n // 2)
            return x * res if n % 2 == 1 else res
        
        res = recursion(x, abs(n))
        return res if n > 0 else 1 / res

File: Data Structures & Algorithms/number-of-provinces/submission-1.js
class Solution {
    /**
     * @param {number[][]} isConnected
     * @return {number}
     */
    findCircleNum(isConnected) {
        let res = 0
        const n = isConnected.length
        const visited = new Array(n).fill(false)

        function dfs(node) {
            visited[node] = true
            for (let nei = 0; nei < n; nei++) {
                if (!visited[nei] && isConnected[node][nei] === 1) {
                    dfs(nei)
                }
            }
        }


        for (let i = 0; i < n; i++) {
            if (!visited[i]) {
                dfs(i)
                res++;
            }
        }
        return res
    }
}
